### PR TITLE
Add TileCloud-chain

### PR DESCRIPTION
**Pull request by sbrunner**
See:
- https://github.com/camptocamp/tilecloud-chain/pull/1200
- https://github.com/camptocamp/helm-geomapfish/pull/19
- https://github.com/camptocamp/mapfish-print-logs/pull/158
- https://github.com/camptocamp/operator-shared-config-manager/pull/51

### CODE REVIEW

File: templates/tilecloudchain-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{{- if .Values.tilecloudchain.enabled -}}
+{{- if .Values.tilecloudchain.ingress.enabled -}}
```
Comment (danduk82) [inline templates/tilecloudchain-ingress.yaml:2]: would something like this work (untested)?
 `{{- if and .Values.tilecloudchain.enabled .Values.tilecloudchain.ingress.enabled -}}`

File: templates/tilecloudchain-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                name: {{ $fullName }}
+                port:
+                  number: {{ $svcPort }}
+              {{- else }}
+              serviceName: {{ $fullName }}
+              servicePort: {{ $svcPort }}
+              {{- end }}
+        {{- end }}
+{{- end }}
+{{- end }}
```
Comment (danduk82) [inline templates/tilecloudchain-ingress.yaml:54]: if the above works, remove one `end` section here

**Review by danduk82 — CHANGES_REQUESTED** [review]
very good job, I love the README, please just change the `if` statement that I have pointed in the comments

File: templates/logs-deployment.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ metadata:
   labels: {{ include "common.labels" ( dict "root" . "service" .Values.logs ) | nindent 4 }}
   annotations: {{- toYaml .Values.logs.annotations | nindent 4 }}
 spec:
-  replicas: {{ .Values.config.replicaCount }}
+  replicas: {{ .Values.logs.replicaCount }}
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply templates/logs-deployment.yaml:9]: was Oups :-)

File: templates/tilecloudchain-ingress.yaml (hunk 1 above)
Comment (sbrunner) [reply templates/tilecloudchain-ingress.yaml]: +1 :-)

File: templates/tilecloudchain-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: /tiles/c2c/health_check
+              port: http
+            initialDelaySeconds: 5
+            timeoutSeconds: 10
```
Comment (danduk82) [inline templates/tilecloudchain-deployment.yaml:54]: I would add a liveness probe for tilecloudchain, the same as the readiness should be ok (i.e. HTTP GET `/tiles/c2c/health_check` )

File: templates/tilecloudchain-ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - hosts:
+        - {{ $ingress_host.host }}
+      secretName: {{ .secretName }}
+  {{- end }}
+{{- end }}
+  rules:
+    - host: {{ $ingress_host.host }}
+      http:
+        paths:
+        {{- range $ingress_host.paths }}
```
Comment (danduk82) [inline templates/tilecloudchain-ingress.yaml:36]: question: would it be better to do as in helm-geoportal ingress, i.e. taking out the paths from the host? Or will the path be dependent of the customer project?

**Review by danduk82 — CHANGES_REQUESTED** [review]
Seems good, please add a liveness probe to the tile-cloud-chain wsgi server

File: templates/tilecloudchain-ingress.yaml (hunk 5 above)
Comment (sbrunner) [reply templates/tilecloudchain-ingress.yaml]: +1 :-)

File: templates/tilecloudchain-deployment.yaml (hunk 4 above)
Comment (sbrunner) [reply templates/tilecloudchain-deployment.yaml:56]: Done

**Review by danduk82 — APPROVED** [review]
good, lgtm

**Pull request merged**